architecture.md

```diff
@@ -0,0 +1,78 @@
+# Ignition Gazebo Architecture
+
+Ignition Gazebo is the application entry-point of Ignition architecture, generally encompassing the use of all other Ignition libraries.
+It is a library itself, but its use case described in this document is as an executable.
+As an executable, it runs a simulation by launching two processes: the backend server process and frontend client process.
+
+The core of Ignition Gazebo is the server-client communication and the various plugins it loads to facilitate the simulation.
+Plugins are loaded at runtime to create a link between other Ignition libraries and Ignition Gazebo so they can exchange information.
+Users can write their own plugins.
+Because they’re loaded at runtime, Ignition does not need to be recompiled to add or remove plugins.
+There are other types of plugins in the scope of Ignition; however, this use case only concerns Ignition Gazebo server and GUI plugins.
+
+Without plugins, Ignition libraries are not “aware” of each other and don’t possess the ability to interact.
+For example, Ignition Physics is independent of Ignition Gazebo, and vice versa.
```

Comment:
It would be good to add as the first sentence of the paragraph, emphasizing that the libraries are independent and modular. It's a selling point.